R/input-check-helpers.R

```diff
@@ -1,114 +1,159 @@
+  )
+  return(invisible(NULL))
+}
+
+#' @title Check input for binary forecast
 #'
-#' @description
-#' Helper function to check inputs for lower-level score functions.
-#' @inheritParams check_predictions
-#' @return NULL
+#' @description Helper function to check whether the input is suitable for
+#' scoring.
+#' @param observed Input to be checked. Should be a factor of length n with
+#' exactly two levels, holding the observed values.
+#' The highest factor level is assumed to be the reference level. This means
+#' that `predicted` represents the probability that the observed value is equal
+#' to the highest factor level.
+#' @param predicted Input to be checked. `predicted` should be a vector of
+#' length n, holding probabilities. Values represent the probability that
+#' the corresponding value in `observed` will be equal to the highest
+#' available factor level.
+#' @importFrom checkmate assert assert_factor
+#' @return Returns NULL invisibly if the check was successful and throws an
+#' error otherwise.
 #' @keywords internal
+check_input_binary <- function(observed, predicted) {
+  # things that need to be checked
+  # - all inputs need to be vectors of the same length
+  # - observed needs to be a factor
+  # - they need to have the same levels
+  # - predicted needs to be numeric between 0 and 1
 
-check_true_values <- function(true_values,
-                              type = c("continuous", "integer", "binary")) {
-  type <- match.arg(type)
-  if (missing(true_values)) {
-    stop("true_values argument is missing")
-  }
+  check_equal_length(observed, predicted)
+  assert_factor(observed, n.levels = 2)
+  levels <- levels(observed)
+  assert(
+    check_numeric_vector(predicted, min.len = 1, lower = 0, upper = 1)
+  )
+  return(invisible(NULL))
+}
+
+check_input_point <- function(observed, predicted) {
+  # things that need to be checked
+  # - all inputs need to be numeric vectors
+  # - they need to have the same length
 
-  if (type == "integer" &&
-      isFALSE(all.equal(true_values, as.integer(true_values)))
-  ) {
-    stop("Some of the true_values are not integers")
+  assert(check_numeric_vector(observed, min.len = 1))
+  assert(check_numeric_vector(predicted, min.len = 1))
+  if (length(observed) != length(predicted)) {
+    stop("observed and predicted need to be",
+         "of same length when scoring point forecasts")
   }
+  return(invisible(NULL))
+}
+
 
-  if (type == "binary" &&
-      isFALSE(all(true_values %in% c(0, 1)))
-  ) {
-    stop("For a binary forecast, all true_values should be either 0 or 1.")
+#' @title Check whether an input is an atomic vector of mode 'numeric'
+#'
+#' @description Helper function
+#' @param x input to check
+#' @param x additional arguments to pass to `check_numeric()`
+#' @importFrom checkmate check_atomic_vector check_numeric
+#' @return Either TRUE if the test is successful or a string with an error
+#' message
+#' @keywords internal
+check_numeric_vector = function(x, ...) {
```

Comment:
use <- - more generally run a linter on this and clean up

Reply:
I suggest doing this a bit later as I already did more work with the current files. It's part of a larger clean up issue: #346

Reply:
What linter are you using? Not that I'm running one regularly, but when I ran one, it didn't flag that...